test/libsolidity/smtCheckerTests/functions/constructor_hierarchy_3.sol

```diff
@@ -0,0 +1,22 @@
+pragma experimental SMTChecker;
+contract C {
+	uint a;
+	constructor(uint x) public {
+		a = x;
+	}
+}
+
+contract B is C {
+	constructor(uint x) public {
+		a = x;
+	}
+}
+
+contract A is B {
+	constructor(uint x) B(x) C(x + 2) public {
+		assert(a == x);
+		assert(a == x + 1);
+	}
+}
```

Comment:
Hm... maybe another one similar to this like:
```
pragma experimental SMTChecker;
contract C {
	uint a;
	constructor(uint x) public {
		a = x;
	}
}

contract B is C {
        uint b;
	constructor(uint x) public {
		b = a + x;
	}
}

contract A is B {
	constructor(uint x) B(x) C(x + 2) public {
		assert(a == x + 2);
		assert(b == x + x + 2);
	}
}
```
With or without letting the ``B`` constructor refer to ``a``...

Reply:
Generally I'm seeing more tests with constructor overriding what others did, verifying that the last one was the correct one, but not really tests for *all* constructors being executed...

Reply:
Ok!

Reply:
Added them